### PR TITLE
Add Rental Entity, Liquibase Changeset, DTOs, Mapper, and Service interface

### PR DESCRIPTION
**Pull request by nklimovych**

### CODE REVIEW

File: src/main/java/online/carsharing/entity/Rental.java [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+    private LocalDate rentalDate;
+
+    @Column(nullable = false, name = "return_date")
+    private LocalDate returnDate;
+
+    @Column(name = "actual_return_date")
+    private LocalDate actualReturnDate;
+
+    @ManyToOne(fetch = FetchType.EAGER)
+    private Car car;
```
Comment (SkillfulDev) [inline src/main/java/online/carsharing/entity/Rental.java:41]:
```suggestion
    @ManyToOne(fetch = FetchType.Lazy)
    private Car car;
```
Maybe you should put Lazy

**Review by SkillfulDev — DISMISSED** [review]

**Review by azzilian — DISMISSED** [review]
GJ! I agree that we should change to lazy somehow

File: src/main/resources/db/changelog/changes/klimovych/insert-rentals-into-rental.yaml [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+                  valueDate: "2024-04-05"
+              - column:
+                  name: actual_return_date
+                  valueDate: "2024-04-05"
+              - column:
+                  name: car_id
+                  value: "2"
+              - column:
+                  name: user_id
+                  value: "1"
```
Comment (HAIOVYI) [inline src/main/resources/db/changelog/changes/klimovych/insert-rentals-into-rental.yaml]: I think this is bad practice to hardcode here user and car id. Also, I believe we must use `preConditions` and `valueComputed`  in our liquibase changeset.
Let's imagine that someone deletes our test user or our test car with the following liquibase changes
then we will not be able to create these rentals and will get an error. For this, it is necessary to check whether these users and cars exist

**Review by HAIOVYI — CHANGES_REQUESTED** [review]

File: src/main/resources/db/changelog/changes/klimovych/insert-rentals-into-rental.yaml (hunk 2 above)
Comment (nklimovych) [reply src/main/resources/db/changelog/changes/klimovych/insert-rentals-into-rental.yaml]: This insert changelog was removed

**Review by HAIOVYI — APPROVED** [review]

**Review by SkillfulDev — APPROVED** [review]

**Pull request merged**